sdks/java/core/src/main/java/org/apache/beam/sdk/io/AvroIO.java

```diff
@@ -760,6 +771,18 @@ private static void validateOutputComponent(String partialFilePattern) {
         + partialFilePattern);
   }
 
+  private static String getBrowseUrl(String filePattern) {
+    IOChannelFactory factory;
+    try {
+      factory = IOChannelUtils.getFactory(filePattern);
```

Comment:
I've updated AvroIO and TextIO to be resilient to bad file schemes when their validation is disabled.  I didn't update sources and sinks, but now I think they probably need the same treatment. I'll work on that now.